### ISSUE TITLE
Enhancements on the SentimentAnalyzer

### ISSUE THREAD

**Issue by arafattehsin** [arafattehsin/CognitiveRocket#4]
As a part of #3 - an idea is floated around keeping a single library with multiple models under the hood. Each model will have a specific dataset(s) on which it is going to be trained for fine grained sentiment analysis. 

User / developer should be able to call the function from by passing a flag (probably Enum?) 

For example;

A user analyzing the sentiments for movie reviews will call the Movie Sentiment Analysis Model. 
A user looking for the customer reviews sentiment analysis will call the Customers Feedback Analysis Model.

**Comment by arafattehsin** [comment]
@kamranayub - your thoughts are most welcome here 😊

**Comment by kamranayub** [comment]
I like that! Some kind of pseudo code here:

```
Sentiments.Predict(text) // Default
Sentiments.Predict(text, SentimentType.MovieReview)
Sentiments.Predict(text, SentimentType.ProductReview)
```

**Comment by kamranayub** [comment]
Just for some more data points, I had some review text I was running through the analyzer:

> I HATE this product, it doesn't do anything I want it to!

And yet it determined the prediction was `Positive` so I do think there's some room for other model training to tweak the results depending on the domain!

One neat thing could be to use the IMDB dataset for tests since it includes the sentiment score (-1 = negative, 0 = neutral, 1 = positive) and test models against it to see how well they match up.

**Comment by arafattehsin** [comment]
Yeah I noticed that as well. You are right there's a room of other model training. I will schedule it soon. Meanwhile, what we can collect are the datasets and finalize the categories.

Movies - [IMDB dataset](https://www.kaggle.com/lakshmi25npathi/sentiment-analysis-of-imdb-movie-reviews/data?select=IMDB+Dataset.csv)
Product Reviews - [Amazon Reviews](https://www.kaggle.com/bittlingmayer/amazonreviews?select=test.ft.txt.bz2)
News / Views - 
.
.
.
.

Above categories are not restricted (they can also be expanded if we want) and also can be trained with more than one dataset. A little to no feature engineering may required which I can handle.
